### ISSUE TITLE
Consider adding a [[sample_position]]

### ISSUE THREAD

**Issue by Kangz** [gpuweb/gpuweb#1994]
This fragment input builtin gives the position inside the pixel of the current fragment. (as a `vec2<f32>` with each coordinate in [0, 1])

## Metal

This is the same as `[[point_coord]]` in Metal.

## Vulkan

This is the same as SPIR-V's `SamplePosition` builtin. It requires the `sampleRateShading` but that's pretty much ubiquitous and already required by `[[sample_index]]` that maps to `SampleID` in SPIR-V.

## D3D12

D3D12 doesn't have an equivalent builtin. However the multisampling pattern is known at pipeline compilation time (and in most implementations I expect it to be known statically). So this builtin could be emulated by using `[[sample_index]]` to index a static array.

**Comment by kvark** [comment]
I know it's good to expose everything that native APIs expose. This particular case isn't very obvious, for the following reasons:
  1. you should be able to get the same value by declaring `[[builtin(position), interpolate(perspective, sample)]] frag_pos: vec4<f32>` input to FS and then getting the `fract(frag_pos)`.
  2. we should instead expose something like `fn sample_location(index: u32) -> vec2<f32>` builtin function, which is supported in all of the APIs, and is ultimately more useful than the `[[sample_position]]` because it can be done for MSAA resolves and other things.
  3. `[[sample_position]]` isn't supported on DX12, anyway

So, we can still expose it, but there is no urgency at all (i.e. easy to work around), and we should probably focus on `sample_position` builtin function instead.

**Comment by Kangz** [comment]
Good point that this is probably not needed. Happy to close this if noone else is interested.

**Comment by kainino0x** [comment]
I think that's what already happens with `[[sample_index]]`?

Agree having a pipeline flag would be nice. #2133
